duolingo.py

```diff
@@ -28,20 +28,84 @@ class DuolingoException(Exception):
 class Duolingo(object):
     USER_AGENT = "Python Duolingo API/{}".format(__version__)
 
-    def __init__(self, username, password):
-        self.username = username
-        self.password = password
-        self.user_url = "https://duolingo.com/users/%s" % self.username
+    _url_base = "https://{host}{base_path}"
+    _url_user_no_auth = "/users?username={username}"
+    _url_login = "/login?fields="
+
+    def __init__(self, username="", password="", host="www.duolingo.com", base_path="/2017-06-30"):
+        self.host = host
+        self.base_path = base_path
+        self.base_url = self._url_base.format(host=self.host, base_path=self.base_path)
+
         self.session = requests.Session()
         self.leader_data = None
-        self.jwt = None
+        self.jwt = None        
+
+        if username:
+            self.username = username
+            if password:
+                self.password = password
+                self._login()
+            self._get_user_data(username)
+        self.voice_url_dict = None
 
-        if password:
-            self._login()
+    def _get_user_data(self, username):
+        """Gets user data with _get_data if logged in
+            and _get_user_data_no_auth otherwise
+        
+        Arguments:
+            username {str} -- username on Duolingo
+        """        
+        self.username = username
+        self.user_url = "https://duolingo.com/users/%s" % self.username # TODO: should be removed
```

Comment:
Looks like you left a todo note here. Does this want to use the self.host?

Reply:
Yes, it should be changed in the whole module. But it would be a very big PR, so I introduced the feature, without breaking current implementation, to make sure it's OK with other developers, then stepwise reimplement there methods. It's actually should be `self.base_url`

Reply:
Okay, that sounds neat. I can swap it all over to use self.host and stuff where appropriate once this is merged then